### DISCUSSION TITLE
Preserving fully transparent yet colored pixels

### DISCUSSION THREAD

**Discussion by Firch** [ImageMagick/ImageMagick#7537] [Help] [answered]
### ImageMagick version

7.1.1-32-Q16-HDRI-x64

### Operating system, version and so on

Windows 10

### Description

Heyo!

Trying to use ImageMagick to automate image upscaling. Here's the conversion I'm using:

`magick convert "file1" -scale 200%% -filter point "file2"`

But during the conversion, some of the transparent colors - the ones that have a value of alpha 0 - are removed completely instead of staying as their respective colors to (#00000000 instead of #whateverRGB00).

Is there any flag I can apply to the conversion so it won't reset the RGB value of a fully transparent pixels?

**Comment by snibgo** [comment, marked as answer]
`magick convert` uses the very old v6 syntax. I suggest removing `convert` so you use just `magick`, which uses the much better v7 syntax.

Your `-filter point` has no effect.

You can extract the alpha, so we have two images. Switch alpha off, scale both images, then copy the alpha (opacity) into the colour image. Like this, Windows CMD format:

```
magick in.png ( +clone -alpha extract ) -alpha off -scale 200% -compose CopyOpacity -composite out.png
```
In a BAT file, double the `%`.

**Reply by Firch** [reply]
Thanks for the answer!

Pretty sure I tried removing `-filter point` before, but it changed the upscale method to bilinear as opposed to nearest neighbor.

**Comment by snibgo** [comment]
`-filter` has no effect on `-scale`. `-filter` does have an effect on `-resize`, provided it is placed *before* the `-resize`.
